modules/auxiliary/scanner/ssl/ssl_version.rb

```diff
@@ -472,7 +473,7 @@ def run_host(ip)
     begin
       ctx = { 'Msf' => framework, 'MsfExploit' => self }
       # Initialize rex-sslscan scanner
-      scanner = Rex::SSLScan::Scanner.new(ip, rport, ctx)
+      scanner = Rex::SSLScan::Scanner.new(ip, rport, ctx, tls_server_name_indication: datastore['SSLServerNameIndication'])
```

Comment:
Once `SSLServerNameIndication` is made optional, would it be possible to do something like this? This would allow the option to explicitly be set from the datastore, but if the user doesn't set it, the RHOSTNAME from RangeWalker would be used by default in the event that they're targeting a host by it's name making it more convenient to use.

```suggestion
      tls_server_name_indication = nil
      tls_server_name_indication = datastore['SSLServerNameIndication'] if tls_server_name_indication.nil? and datastore['SSLServerNameIndication'].present?
      tls_server_name_indication = datastore['RHOSTNAME'] if tls_server_name_indication.nil? and datastore['RHOSTNAME'].present?
      scanner = Rex::SSLScan::Scanner.new(ip, rport, ctx, tls_server_name_indication: tls_server_name_indication)
```

Reply:
That's roughly the approach taken elsewhere too:

https://github.com/rapid7/metasploit-framework/blob/0bc993b9cab0c8961cbfd94c9148e36a659a6d27/lib/msf/core/exploit/remote/tcp.rb#L105

It's maybe worth noting that Pro doesn't currently use the rhost walker for IP expansion, so RHOSTNAME wouldn't be set automatically by Pro with its current implementation

Reply:
> Once `SSLServerNameIndication` is made optional, would it be possible to do something like this? This would allow the option to explicitly be set from the datastore, but if the user doesn't set it, the RHOSTNAME from RangeWalker would be used by default in the event that they're targeting a host by it's name making it more convenient to use.

@smcintyre-r7 thank you for the proposal, i have added this to the code.

Reply:
> That's roughly the approach taken elsewhere too:
> 
> https://github.com/rapid7/metasploit-framework/blob/0bc993b9cab0c8961cbfd94c9148e36a659a6d27/lib/msf/core/exploit/remote/tcp.rb#L105
> 
> It's maybe worth noting that Pro doesn't currently use the rhost walker for IP expansion, so RHOSTNAME wouldn't be set automatically by Pro with its current implementation

@adfoster-r7 i have created a task for it